vspace/src/main/java/edu/asu/diging/vspace/core/factory/impl/SpaceFormFactory.java

```diff
@@ -0,0 +1,61 @@
+package edu.asu.diging.vspace.core.factory.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.factory.ISpaceFormFactory;
+import edu.asu.diging.vspace.core.model.IExhibition;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.services.IExhibitionManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+import edu.asu.diging.vspace.web.staff.forms.SpaceForm;
+
+@Service
+public class SpaceFormFactory implements ISpaceFormFactory {
+    
+    @Autowired
+    private IExhibitionManager exhibitionManager;
+    
+    @Autowired
+    private ISpaceManager spaceManager;
+    
+    @Autowired
+    private LocalizedTextFormCreation localizedTextFormCreation;
+
+    /**
+     * 
+     * Creates new space form object
+     * @param space
+     * @return
+     */
+    @Override
+    public SpaceForm createNewSpaceForm(ISpace space) {
+        SpaceForm spaceForm = new SpaceForm();
+        spaceForm.setName(space.getName());
+        spaceForm.setDescription(space.getDescription());
+        
+        IExhibition startExhibtion = exhibitionManager.getStartExhibition();
+        
+        spaceForm.setDefaultName(localizedTextFormCreation.createLocalizedTextForm(startExhibtion.getDefaultLanguage(), space.getSpaceNames()));
+        spaceForm.setDefaultDescription(localizedTextFormCreation.createLocalizedTextForm(startExhibtion.getDefaultLanguage(), space.getSpaceDescriptions()));
```

Comment:
I think you can just move these two lines into the loop below into an else block. No need to iterate over all languages again to get the default language when you are already iterating over it below.